pkg/features/kube_features.go

```diff
@@ -911,7 +911,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 	DisableAcceleratorUsageMetrics:                 {Default: true, PreRelease: featuregate.Beta},
 	HPAContainerMetrics:                            {Default: false, PreRelease: featuregate.Alpha},
 	SizeMemoryBackedVolumes:                        {Default: true, PreRelease: featuregate.Beta},
-	ExecProbeTimeout:                               {Default: true, PreRelease: featuregate.GA}, // lock to default and remove after v1.22 based on KEP #1972 update
+	ExecProbeTimeout:                               {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove after v1.24
```

Comment:
No objections locking this gate, but I personally don't see a lot of harm is keeping it unlocked for longer either. The amount of code required on our end to maintain this is very small and manageable IMO. 

Waiting 4 releases actually seems pretty reasonable to lock this, but just curious if there's a particular reason for locking it now instead of waiting a little longer, are we confident that most folks have made the appropriate fixes?

Reply:
Good question. It's because the `ExecProbeTimeout=false` configuration doesn't actually work for non-dockershim (of course there is no other CRI abstraction going forward. Specifically, this scenario will engage an edge case bug that yields a liveness probe result of always success:

```
    args:
    - /bin/sh
    - -c
    - while true; do rm -f /tmp/healthy; sleep 5; done
    livenessProbe:
      exec:
        command: ['sh', '-c', 'sleep 5 && test -f /tmp/healthy']
      initialDelaySeconds: 1
      periodSeconds: 1
      failureThreshold: 1
      timeoutSeconds: 1
```

I've tried to describe this in the PR description as well. tl;dr the `ExecProbeTimeout=false` "preserve the pre-1.20 behavior" solution made sense for dockershim, because it simply disabled exec probe timeout enforcement while preserving the integrity of the probe response itself (this is what the pre-1.20 behavior was, in summary). But for non-dockershim, the pre-1.20 behavior was pathological (see the above example) and so we don't actually want to preserve it. Rather, we want to force folks onto this 1.20+ happy path, because otherwise their exec probe enforcement is actually buggy (any probe that takes longer than 1 second by default will alwyas succeed, regardless of whether or not the probe command returns a non-zero exit code).

Reply:
> Rather, we want to force folks onto this 1.20+ happy path, because otherwise their exec probe enforcement is actually buggy (any probe that takes longer than 1 second by default will alwyas succeed, regardless of whether or not the probe command returns a non-zero exit code).

Aren't we already doing this by having the feature gate on by default? Or are there non-dockershim users that are incorrectly turning it off thinking they want to perserve the old behavior?

Reply:
I'm not sure that those non-dockershim users are aware that this feature-gate isn't designed for them, as this pathology was only discovered after the 1.20 release landed and the general guidance about why you might want to use it was issued (that guidance AFAIK did not emphasize _For dockershim Scenarios Only!_).